linchpin/cli/__init__.py

```diff
@@ -355,7 +355,7 @@ def lp_down(self, pinfile, targets=(), run_id=None):
         pass
 
 
-    def lp_up(self, targets=(), run_id=None, tx_id=None, inv_f="cfg"):
+    def lp_up(self, targets=(), run_id=None, tx_id=None, inv_f="cfg", flags={}):
```

Comment:
Why not use the context (self.ctx) here? It seems `flags` is just another thing to pass. The point of the ctx is that it is the ONE thing that is available in most (if not all) classes and should contain this data. In fact, the `inv_f` thing is likely something that could go through the ctx as well.

Plus, as I see above, the settings are already in the linchpin.constants/linchpin.conf. Adding this flag to the [runcli method](https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/linchpin/shell/__init__.py#L145) as shown in this link. Then you can just access the ctx values as necessary.

Reply:
Totally agree with you . But I would not encourage adding flag to runcli method since the command structure might change with would effect usability of flags
``` 
linchpin -vvvv --creds-path --run-hooks-on-failure up 
```
The below looks more intutive and easy to remember since hooks happen after provisioning
```
linchpin -vvvv --creds-path up --run-hooks-on-failure
```
However, as you have suggested one can always set the context e_vars or cfg vars within the function. which can be done. And also avoids passing another arguments
Any thoughts ?

Reply:
Good point. I may have misspoke about where to put the code. I do think in the up/down methods is more likely the better location. I'm happy with wherever you think is best, just that in the latter case, you'd have to add the options twice.